packages/hydrooj/src/model/contest.ts

```diff
@@ -718,6 +721,8 @@ async function _updateStatus(
     }, 'rid');
     const journal = _getStatusJournal(tsdoc);
     const stats = RULES[tdoc.rule].stat(tdoc, journal);
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    if (status === STATUS.STATUS_ACCEPTED && tdoc.balloon && !isLocked(tdoc)) await addBalloon(tdoc.domainId, tdoc._id, pid, uid);
```

Comment:
whether send a balloon should depends on the record submission time, not current time.